File: lab6_dist_tstudent.R
# Title     : Distribuição T de Student
# Objective : Uso de R para manipular dados que atendam à distribuição T de Student
# Created by: accol
# Created on: 25/05/2020

# Ferramentas que estão disponíveis no R
# P[X < x]
# pt() >- se for uma probabilidade do tipo menor que
# P[X > x] = lower.tail = FALSE >- se for uma probabilidade do tipo maior que

# Prática 1: cientistas de dados ganham em média 75,00 a hora, tamanho da amostra igual 9 com desvio
# padrão da amostra igual 10. Pergunta-se qual a probabilidade de selecionar um cientista de dados e
# o salário for menor que 80,00 a hora? Sabemos da tabela T que t = 1,5, assim temos:
# pt() >- exige dois parâmetros no caso em que temos um problema do tipo menor que >- o valor de t
# (extraído da tabela) e tø (n -1) = 8
probabilidade1 = pt(1.5, 8)
print(probabilidade1)

# Prática 2: qual a probabilidade de selecionar um cientista de dados com salário superior a 80,00?
# Note que se trata de um problema maior que >- devemos usar o parâmetro lower.tail
probabilidade2 = pt(1.5, 8, lower.tail=FALSE)
print(probabilidade2)

# Lembrando que a probabilidade é cumulativa a partir da esquerda, poderíamos encontrar o mesmo
# resultado fazendo:
probextra = 1 - probabilidade1
print(probextra)
# Caso queira confirmar o resultado, poderá verificar se a soma das duas probabilidades, no caso,
# dos problemas acima ser igual a 1 (100%)
probcem = probabilidade1 + probabilidade2
print(probcem)

File: lab6_dist_tstudent.py
# Title     : Distribuição T de Student
# Objective : Uso de Python para manipular dados que atendam à distribuição T de Student
# Created by: accol
# Created on: 25/05/2020

# Ferramentas que estão disponíveis no Python


from scipy.stats import t 


# Problema 1: Media de salário dos cientistas de dados = R$ 75,00 por hora. A amostra é de 9 funcionários e o desvio padrão da amostra é 10
# Qual a probabilidade de selecionar um cientista de dados com salário com salário inferior a R$ 80,00 a hora?


# Estamos trabalhando com problemas do tipo menor que, logo usaremos a função cdf() >- usaremo t = 1,5 extraído da tabela e os demais parâmetros conforme enunciado tø = (9 -1) = 8

prob1 = t.cdf(1.5, 8)
print('\nA probabilidade de selecionar um cientista de dados com salário inferior a 80,00 é: ', prob1)

# Problema 2: Media de salário dos cientistas de dados = R$ 75,00 por hora. A amostra é de 9 funcionários e o desvio padrão da amostra é 10
# Qual a probabilidade de selecionar um cientista de dados com salário com salário maior de R$ 80,00 a hora?

# Estamos trabalhando com problemas do tipo maior que, logo usaremos a função sf() >- usaremo t = 1,5 extraído da tabela e os demais parâmetros conforme enunciado tø = (9 -1) = 8

prob2 = t.sf(1.5, 8)
print('\nA probabilidade de selecionar um cientista de dados com salário superior a 80,00 é: ', prob2)

# Podemos obter o mesmo resultado em % multiplicando por 100
print('\nA probabilidade de selecionar um cientista de dados com salário superior a 80,00 é: ', prob2 * 100, '%')

# Verificando os resultados com base no conceito de que a probabilidade é cumulativa a partir da esquerda >- deve ser igual a unidade
print('\nA probabilidade total dos slarios cientista de dados é: ', prob1 + prob2)
# Podemos encontra o valor de prob2 realizar subtraindo prob1 da unidade, observe
print('\nA probabilidade de selecionar um cientista de dados com salário superior a 80,00 é: ', (1 - prob1) * 100, '%')
